src/controllers/post.js

```diff
@@ -33,7 +33,15 @@ export const create = async (req, res) => {
 }
 
 export const getAll = async (req, res) => {
+  const id = parseInt(req.user.id)
+  let condition
+  if (req.user.role === 'STUDENT') {
+    condition = {
+      OR: [{ isPrivate: false }, { userId: id }]
+    }
+  }
   const posts = await dbClient.post.findMany({
+    where: condition,
```

Comment:
I think this would be better if you attached the where clause to the query instead of storing it in a variable that might be undefined.

```js
const query = {
  skip: 0,
  take: 100,
  orderBy: {...}
}

if (req.user.role === 'STUDENT') {
  query.where = {
      OR: [{ isPrivate: false }, { userId: id }]
   }
}

dbClient.post.findMany(query)
```

Reply:
Updating it now, thanks